packages/provider/index.js

```diff
@@ -32,10 +31,7 @@ module.exports = {
     return provider;
   },
 
-  testConnection: function(options) {
-    const provider = this.getProvider(options);
-    const interfaceAdapter = new InterfaceAdapter();
-    const web3 = new Web3Shim({ provider });
+  testConnection: function(web3, interfaceAdapter) {
```

Comment:
By removing the logic in `getProvider` you will be removing functionality. For example, what happens when someone supplies the provider as a function in the Truffle config? Am I missing something or is that handled somewhere?

Reply:
I guess this just means that the provider needs to be provided while instantiating the web3 passed to this function. I guess the real question is, is that addressed in the code calling this function?

Reply:
[`config.provider`](https://github.com/trufflesuite/truffle/pull/2504/files#diff-69a1d20ea0c7b0784b313148d5a23169R20) will have already hit [this getter](https://github.com/trufflesuite/truffle/blob/develop/packages/config/src/configDefaults.ts#L197-L206) by the time it's passed to `Environment.detect`.

Reply:
I'm also in favor of going back to the drawing board and reverting the original PR to remove a potential blocker for release.

Reply:
(in order to provide time to re-think logging, where we want to test the connection, etc)

Reply:
So I don't think we really mind testing it a bunch of times but I suppose the logging is wonky. We can remove it for now, it just seems weird to have a 20 second timeout waiting for the response and not let the user know what is going on. Let's talk about it at/after dev-sync today.

Reply:
I really don't think it makes sense to explicitly call `getProvider` when `config.provider` has the aforementioned getter, esp if we keep the `testConnection` call in `Environment.detect`.